### ISSUE TITLE
Add a reverse clamp method

### ISSUE THREAD

**Issue by advent94** [godotengine/godot-proposals#8995]
### Describe the project you are working on

I am making Pokemon Blue replica but it doesn't really matter.

### Describe the problem or limitation you are having in your project

I wanted to loop through menu options and once I scroll down last object, I wanted to point on first one. And if I wanted to scroll up from first element, I wanted to end up on last. I was looking for reverse clamp, checked wrap (it's not it)  and didn't find anything like that. I might have missed something so I apologize if I have overseen it. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Reverse clamp. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

```
 func rclamp(a, min, max):
 if (a < min):
     a = max
 elif (a > max):
     a = min
 return a
```

can be same for float or any object type that has comparison if necessary

### If this enhancement will not be used often, can it be worked around with a few lines of script?

```
 func rclamp(a, min, max):
 if (a < min):
     a = max
 elif (a > max):
     a = min
 return a

```
### Is there a reason why this should be core and not an add-on in the asset library?

Just an idea that more people might want to have need for it, just like people need clamp sometimes. It's specific but pretty central for GUI or maybe for example something like shooting right border to get bullets appear on right border. Stuff like that.

**Comment by Calinou** [comment]
> I was looking for reverse clamp, checked wrap (it's not it) 

Can you describe why `wrap()` doesn't suit your use case? To my knowledge, this is exactly what it's designed for.

**Comment by advent94** [comment]
Certainly. So wrap(15, 0, 10) would return value of 5. Clamp keeps value between min and max once value is lower or higher than limiters.  For rclamp(15, 0, 10) we would receive value of 0. We went through limiter (10) and we do want to "limit" our value to max/min but also reverse that outcome. It's additional protection in case anyhow our value would be higher than 1 to iterate through elements. We make sure that under this one function, we stick to limiters but in reverse. It's quite specific but I thought it could be helpful addition to Godot's toolset.
